### PR TITLE
Updating production guide with information about switching from ipv4 to ipv6

### LINKED ISSUE #10482: IPv6 listener support
### Description

Keycloak 17.0.0 defaults to listening on IPv4 addresses only (0.0.0.0).  If I try to listen on IPv6, it says the protocol family is unavailable:

```
# sudo -u keycloak /opt/keycloak/bin/kc.sh start --http-host="[::]"
...
2022-02-25 14:50:38,518 ERROR [org.keycloak.quarkus.runtime.cli.ExecutionExceptionHandler] (main) ERROR: Unable to start HTTP server
2022-02-25 14:50:38,519 ERROR [org.keycloak.quarkus.runtime.cli.ExecutionExceptionHandler] (main) ERROR: java.net.SocketException: Protocol family unavailable
2022-02-25 14:50:38,520 ERROR [org.keycloak.quarkus.runtime.cli.ExecutionExceptionHandler] (main) ERROR: Protocol family unavailable
```

Environment: Ubuntu 20.04, regular (not Wildfly) distribution.

### Discussion

_No response_

### Motivation

It is 2022, and it should be possible to run servers either in IPv6-only environments, or dual-stack where the service is accessible via both protocols.

### Details

_No response_

### LINKED ISSUE #11979: java.net.UnknownHostException in database connection when using IPv6 only hostname
### Describe the bug

When using a hostname that has no `A` record, but `AAAA` records, Keycloak will fail to boot with `org.postgresql.util.PSQLException`, caused by `java.net.UnknownHostException`.

Workaround is to set `JAVA_OPTS` to exclude `-Djava.net.preferIPv4Stack=true`, see #10482

### Version

18.0.0

### Expected behavior

_No response_

### Actual behavior

_No response_

### How to Reproduce?

fly.io provides a PaaS experience with managed postgres where the internal network is entirely IPv6 based - very easy to reproduce there without needing to set up a proper IPv6 stack

### Anything else?

_No response_

### PR DESCRIPTION
**Pull request by pedroigor**
Closes #11979
Closes #10482

### CODE REVIEW

**Comment by pedroigor** [comment]
@keycloak/cloud-native-team

File: docs/guides/src/main/server/configuration-production.adoc [hunk 1]
```diff
@@ -36,4 +36,19 @@ Keycloak runs on top of JGroups and Infinispan, which provide a reliable, high-a
 
 To find out more about using multiple nodes, the different caches and an appropriate stack for your environment, see the <@links.server id="caching"/> guide.
 
+== Stack and address preference
+
+The system properties `java.net.preferIPv4Stack` and `java.net.preferIPv6Addresses` are used to configure the JVM for use with IPv4 or IPv6 addresses.
+
+With Keycloak, in order to run using IPv4 addresses, you need to specify `java.net.preferIPv4Stack=true`; in order to run with IPv6 addresses,
```
Comment (DGuhr) [inline docs/guides/src/main/server/configuration-production.adoc:43]:
```suggestion
Keycloak uses the IPv4 address stack by default. In order to run with IPv6 addresses,
```

Comment (DGuhr) [reply docs/guides/src/main/server/configuration-production.adoc:43]: @pedroigor This is the default, so there is not really a "need to specify" it, right?

File: docs/guides/src/main/server/configuration-production.adoc [hunk 2]
```diff
@@ -36,4 +36,19 @@ Keycloak runs on top of JGroups and Infinispan, which provide a reliable, high-a
 
 To find out more about using multiple nodes, the different caches and an appropriate stack for your environment, see the <@links.server id="caching"/> guide.
 
+== Stack and address preference
```
Comment (abstractj) [inline docs/guides/src/main/server/configuration-production.adoc:39]: I'd suggest `Configure Keycloak Server with IPv6 or IPv4`, but this is not a dealbreaker to merge, only a suggest to make things easier to search.

**Review by abstractj — DISMISSED** [review]

File: docs/guides/src/main/server/configuration-production.adoc (hunk 1 above)
Comment (pedroigor) [reply docs/guides/src/main/server/configuration-production.adoc]: Changed.

File: docs/guides/src/main/server/configuration-production.adoc (hunk 2 above)
Comment (pedroigor) [reply docs/guides/src/main/server/configuration-production.adoc]: Done.

**Review by DGuhr — APPROVED** [review]

**Review by abstractj — APPROVED** [review]

**Pull request merged**